inginious/frontend/plugins/user_management/update_user_data.py

```diff
@@ -0,0 +1,214 @@
+    """ change the email of a user """
+    user_filter_users = {
+        "username": username
+    }
+    new_email_users = {
+        "$set": {"email": param}
+    }
+
+    ans = collections_manager.update_collection("users", user_filter_users, new_email_users)
+    return ans.modified_count
+
+
+def change_username(username, new_username, collection_manager, collection_name_list):
+    """ change the username of a user """
+    collection_information_list = read_collections_info_file()
+    default_collection_information = [{"path": "username", "index_array": []}]
+    count = 0
+    for collection_name in collection_name_list:
+        if collection_name in collection_information_list:
+            information = collection_information_list[collection_name]
+        else:
+            information = default_collection_information.copy()
+            # index_array indicates witch fields in the collection are arrays. By default there is just username
+            # therefore if username is an array the index is 0.
+            information[0]["index_array"] = [0] if username_is_array(collection_name, collection_manager) else []
+
+        count += _crete_update_to_change_username_collection(username, new_username, collection_name, information,
+                                                             collection_manager)
+    return count
+
+
+def close_user_sessions(username, collection_manager):
+    """ Close the open sessions of a user """
+    update_filter = {
+        "data.username": username
+    }
+    close_sessions = {
+        "$set": {
+            "data.loggedin": False
+        }
+    }
+
+    return collection_manager.update_many_in_collection("sessions", update_filter, close_sessions)
+
+
+def add_block_user(username, collection_manager):
+    """ Add the field block to a user's document in users collection to indicate that the user can't do logging"""
+    user_filter = {
+        "username": username
+    }
+    new_name = {
+        "$set": {"block": True}
+    }
+
+    return collection_manager.update_collection("users", user_filter, new_name)
+
+
+def unlock_user(username, collection_manager):
+    """ Removes the block field from the user's document """
+    user_filter = {
+        "username": username
+    }
+    new_name = {
+        "$unset": {"block": ""}
+    }
+
+    return collection_manager.update_collection("users", user_filter, new_name)
+
+
+def is_array(collection_information):
+    """ return a boolean value to determinate if the field that is contents the username is an array """
+    path = collection_information["path"]
+    index_list = sorted(collection_information["index_array"])
+    if not index_list:
+        return False
+
+    num_fields = len(_get_field_end_index_in_path(path))
+    last_index_path = num_fields - 1
+    return last_index_path == index_list[-1]
+
+
+def _crete_update_to_change_username_collection(username, new_username, coll_name, collection_info, collection_manager):
```

Comment:
Not sure why this function has this weird name and the spelling wrong. Here is a suggestion
```suggestion
def _update_username_in_all_collections(username, new_username, coll_name, collection_info, collection_manager):
```